[gitlab_runner] Replace GitLab APT repo key

imrejonk opened the pull request:
GitLab.com is using a new package repository signing key:
https://about.gitlab.com/releases/2020/03/30/gpg-key-for-gitlab-package-repositories-metadata-changing/

These changes make Ansible install the new key and remove the old one.

imrejonk committed:
[gitlab_runner] Replace GitLab APT repo key

GitLab.com is using a new package repository signing key:
https://about.gitlab.com/releases/2020/03/30/gpg-key-for-gitlab-package-repositories-metadata-changing/

These changes make Ansible install the new key and remove the old one.

nqb commented:
Hello,

I try to use code of this PR with `debops` 1.2.1. New key has been installed but old key still presents.

Ansible task doesn't generate a `changed` state for oldest key:
```
TASK [debops.keyring : Import specified GPG keys to APT keyring] **************************************************************************************************************************************************
changed: [runner1] => (item={'id': 'F6403F6544A38863DAA0B6E03F01618A51312F3F', 'repo': 'deb https://packages.gitlab.com/runner/gitlab-runner/debian/ buster main', 'state': 'present'})
changed: [ci] => (item={'id': 'F6403F6544A38863DAA0B6E03F01618A51312F3F', 'repo': 'deb https://packages.gitlab.com/runner/gitlab-runner/debian/ buster main', 'state': 'present'})
ok: [ci] => (item={'id': '1A4C919DB987D435939638B914219A96E15E78F4', 'state': 'absent'})
ok: [runner1] => (item={'id': '1A4C919DB987D435939638B914219A96E15E78F4', 'state': 'absent'})
```

Key is still presents in `apt-key`:
```console
# apt-key list 1A4C919DB987D435939638B914219A96E15E78F4
pub   rsa4096 2015-04-17 [SC] [expired: 2020-04-15]
      1A4C 919D B987 D435 9396  38B9 1421 9A96 E15E 78F4
uid           [ expired] GitLab B.V. (package repository signing key) <packages@gitlab.com>
```

imrejonk commented:
@nqb Looks like the apt_key Ansible module does not touch expired keys, intentionally: https://github.com/ansible/ansible/pull/21539. Having a way to manage those keys as well would be nice. There is an issue created about this: https://github.com/ansible/ansible/issues/25070

The key was still valid when I tested my changes :roll_eyes:

nqb commented:
@imrejonk, thanks for your feedback. If I understand correctly, we can remove old key from `ansible/roles/gitlab_runner/defaults/main.yml`

imrejonk committed:
[gitlab_runner] Do not remove expired APT key

Updating or removing expired APT keys is not possible with the apt_key
module: https://github.com/ansible/ansible/issues/25070

imrejonk commented:
@nqb Done :)
For now at least, maybe it will become possible to manage expired APT keys in the future.

drybjed reviewed (approved):
Nicely done. I'm not sure why Travis-CI doesn't want to test this PR, I guess it's some miscommunication. I'll check it locally before merging.
